pallene/checker.lua

```diff
@@ -482,6 +482,11 @@ function Checker:check_stat(stat)
                 types.tostring(itertype), types.tostring(iteratorfn._type))
         end
 
+        if iteratorfn._tag == "ast.Exp.CallFunc" then
+            local iterfn_var = iteratorfn.exp.var._name
+            stat.is_ipairs = iterfn_var._tag == "checker.Name.Builtin" and iterfn_var.name == "ipairs"
```

Comment:
I think we can look at the `_name` annotation in the to_ir.lua. The checker doesn't need to know about this optimization that we are doing with ipairs.